src/Bowling.js

```diff
@@ -0,0 +1,104 @@
+var Scorecard = function() {
+  this.frames = [[],[],[],[],[],[],[],[],[],[]];
+  this.currentFrame = 0;
+  this.currentBowl = 0;
+  this.ballNumber = 1;
+  this.allBalls = {};
+  this.isTheGameOver = false;
+  this.finalFramePoints = 0;
+};
+
+Scorecard.prototype.addPoints = function(points) {
+  if (this.isTheGameOver === true) {
+    return;
+  } else if (this.currentFrame === 9) {
+    this.finalFrame(points);
+  } else {
+    this.updateStrikePoints(points);
+    this.updateSparePoints(points);
+    this.frames[this.currentFrame][this.currentBowl] = points;
+    this.recordBallNumber();
+    this.updateBowlNumber();
+    this.updateFrameNumber();
+    this.ballNumber += 1;
+  }
+};
```

Comment:
very large method - could be a clue to extract into a more object-oriented application. The method name is addPoints, so the function should just add points, instead of doing all this other stuff.